mlflow/langchain/utils.py

```diff
@@ -243,27 +243,31 @@ def _get_supported_llms():
     import langchain.chat_models
     import langchain.llms
 
-    llms = {langchain.llms.openai.OpenAI, langchain.llms.huggingface_hub.HuggingFaceHub}
+    supported_llms = set()
 
-    if hasattr(langchain.llms, "Databricks"):
-        llms.add(langchain.llms.Databricks)
+    def try_adding_llm(module, class_name):
+        if cls := getattr(module, class_name, None):
+            supported_llms.add(cls)
 
-    if hasattr(langchain.llms, "Mlflow"):
-        llms.add(langchain.llms.Mlflow)
-
-    if hasattr(langchain.chat_models, "ChatDatabricks"):
-        llms.add(langchain.chat_models.ChatDatabricks)
+    def safe_import_and_add(module_name, class_name):
+        """Add conditional support for `partner` and `community` APIs in langchain"""
+        try:
+            module = importlib.import_module(module_name, class_name)
```

Comment:
do we need class_name?

Reply:
not at all. good catch. That's a superfluous addition